### PR TITLE
JavaScript capstone project - API-based webapp

### PR DESCRIPTION
**Pull request by Log-benjamin**
### The Contributors to the project are 

- [ ] Binyam Tegene @Log-benjamin 
- [ ] Abbas Aljuaifari @Abbas-Aljuaifari 

### Project Objectives we applied

- Used JavaScript to make websites dynamic and build basic single-page apps.
- Used ES6 syntax.
- Used ES6 modules.
- Used callbacks and promises.
- Used Webpack.
- Applied JavaScript best practices and language style guides in code.
- Used AAA pattern for unit tests.
- Wrote unit tests for a JavaScript app.
- Followed Gitflow.
- Solved git conflicts.
- Implemented Sending and receiving data from an API.
- Used API documentation.
- Understand and use JSON.
- Made JavaScript code asynchronous.
- Perform a code review for a team member.

### Sneak Peak
image

image

image

### CODE REVIEW

File: src/modules/loder.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  });
+
+  cardContainer.addEventListener('click', (e) => {
+    e.preventDefault();
+    if (e.target.className === 'commentBtn') {
+      setTimeout(async () => {
+        const comments = await fetchComments(e.target.id);
+        displayComments(comments);
+        displayCommentCount(comments.length);
+      }, 500);
```
Comment (anagudelogu) [inline src/modules/loder.js:55]: - [ ] **[OPT]** I don't know if this delay is intentional, but please take in mind that it doesn't provide a good experience for the end user! I would suggest not using delays unless you know entirely that you need them! 🤓 🔧

File: src/modules/loder.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      postComment(countryName, userName, userComment);
+      document.getElementById('userName').value = '';
+      document.getElementById('userComment').value = '';
+
+      setTimeout(async () => {
+        const comments = await fetchComments(countryName);
+        displayComments(comments);
+        displayCommentCount(comments.length);
+      }, 500);
+    }
```
Comment (anagudelogu) [inline src/modules/loder.js:83]: - [ ] **[OPT]** I don't know if this delay is intentional, but please take in mind that it doesn't provide a good experience for the end user! I would suggest not using delays unless you know entirely that you need them! 🤓 🔧

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @Log-benjamin @Abbas-Aljuaifari, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 


## Optional suggestions

_Every comment with the [OPT] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/loder.js (hunk 1 above)
Comment (Log-benjamin) [reply src/modules/loder.js:55]: Thank you for the review . We will look through our code to make the app more effeicient.

**Pull request merged**